packages/nest/test/test-app.ts

```diff
@@ -81,15 +91,21 @@ export class OpenFeatureController {
         const userId = request.header('x-user-id');
         return userId
           ? {
-              targetingKey: userId,
-            }
+            targetingKey: userId,
+          }
           : undefined;
       },
     })
     feature: Observable<EvaluationDetails<number>>,
   ) {
     return feature.pipe(map((details) => this.testService.serviceMethod(details)));
   }
+
+  @Get('/dynamic-context/service')
+  public handleDynamicContextForService() {
+    return this.testService.injectedFlagWithContextMethod("dummy");
+  }
```

Comment:
This is, why I think service method decorators could be hard to do with good ergonomics.
Having these makes callers of the service methods having to provide "dummy" arguments as service methods are typically called by a controller or other services, while controllers are called by Nest.
My feeling is that FeatureFlag injection will only work good for injecting FeatureFlags into request scoped services via the constructor because this is the only guaranteed to be called by Nest.

Reply:
I understand your concern. It's not really a great experience from the caller side.

> My feeling is that FeatureFlag injection will only work good for injecting FeatureFlags into request scoped services via the constructor because this is the only guaranteed to be called by Nest.
I'm not a big fan of request scoped services but we cant try it and see what we get.

Reply:
> I'm not a big fan of request scoped services but we cant try it and see what we get.

Generally I thinks they can be a good option, but we could also just decide that services just can not get injected feature flags nicely. 
I dont know, we should play around with and see if we find a good option or if it is not possible to do nicely.